File: LeetCode/Hard/4 MedianOfTwoSortedArrays/4MedianOfTwoSortedArrays.py
from typing import List

class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        m, n = len(nums1), len(nums2)
        merged = []
        i = j = 0

        while i < m and j < n:
            if nums1[i] < nums2[j]:
                merged.append(nums1[i])
                i += 1
            else:
                merged.append(nums2[j])
                j += 1

        while i < m:
            merged.append(nums1[i])
            i += 1

        while j < n:
            merged.append(nums2[j])
            j += 1

        total = m + n
        if total % 2 == 1:
            return float(merged[total // 2])
        else:
            return (merged[total // 2 - 1] + merged[total // 2]) / 2.0

if __name__ == "__main__":
    sol = Solution()
    nums1 = [1, 3]
    nums2 = [2]

    mediana = sol.findMedianSortedArrays(nums1, nums2)
    print(mediana)

File: LeetCode/Hard/4 MedianOfTwoSortedArrays/4MedianOfTwoSortedArrays.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        int m = nums1.size();
        int n = nums2.size();
        vector<int> merged;
        int i = 0, j = 0;

        while (i < m && j < n) {
            if (nums1[i] < nums2[j]) {
                merged.push_back(nums1[i]);
                i++;
            } else {
                merged.push_back(nums2[j]);
                j++;
            }
        }

        while (i < m) {
            merged.push_back(nums1[i]);
            i++;
        }

        while (j < n) {
            merged.push_back(nums2[j]);
            j++;
        }

        int total = m + n;
        if (total % 2 == 1) {
            return merged[total / 2];
        } else {
            return (merged[total / 2 - 1] + merged[total / 2]) / 2.0;
        }
    }
};

int main() {
    Solution sol;
    vector<int> nums1 = {1, 3};
    vector<int> nums2 = {2};

    double mediana = sol.findMedianSortedArrays(nums1, nums2);
    cout << mediana << endl;

    return 0;
}